Examples/HttpAnalyzer/main.cpp

```diff
@@ -552,10 +559,25 @@ int main(int argc, char* argv[])
 	if (readPacketsFromPcapFileName == "" && interfaceNameOrIP == "")
 		EXIT_WITH_ERROR("Neither interface nor input pcap file were provided");
 
+	//get the port
+	std::istringstream is(port);
+	unsigned short nPort = -1;
+	is >> nPort;
+	if (is.fail())
+	{
+		printUsage();

```

Comment:
in case of failure we should print an error message to the user. You can use the macro `EXIT_WITH_ERROR`